node/src/signature_mngr.rs

```diff
@@ -48,11 +56,56 @@ struct SignatureManager {
 struct SetKey(SK);
 struct Sign(Vec<u8>);
 
+fn persist_extended_sk(extended_sk: ExtendedSK) -> impl Future<Item = (), Error = failure::Error> {
+    let extended_secret_key = ExtendedSecretKey::from(extended_sk);
+
+    storage_mngr::put(&EXTENDED_SK_KEY, &extended_secret_key).inspect(|_| {
+        log::debug!("Successfully persisted the extended secret key into storage");
+    })
+}
+
 impl Actor for SignatureManager {
     type Context = Context<Self>;
 
-    fn started(&mut self, _ctx: &mut Self::Context) {
+    fn started(&mut self, ctx: &mut Self::Context) {
         log::debug!("Signature Manager actor has been started!");
+
+        storage_mngr::get::<_, ExtendedSecretKey>(&EXTENDED_SK_KEY)
+            .and_then(move |extended_sk_from_storage| {
+                extended_sk_from_storage.map_or_else(
+                    || -> Box<dyn Future<Item = (), Error = failure::Error>> {
+                        log::warn!("No extended secret key in storage");
```

Comment:
We may inform the user here that nothing bad actually happened, as we will just be generating and persisting a new secret key.
```rust
log::info("Generating and persisting a new master key for this node")
```